src/System.Security.Permissions/src/System/Diagnostics/EventLogPermissionEntryCollection.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+namespace System.Diagnostics
+{
+    [Serializable]
+    public class EventLogPermissionEntryCollection : CollectionBase
+    {
+        internal EventLogPermissionEntryCollection() { }
```

Comment:
I modified the test to filter out classes with internal and protected default constructors. This seems to work, but I'm unsure if its what is desired. 

` IEnumerable<Type> serializableTypes = typeof(CodeAccessPermission).Assembly
                .GetTypes()
                .Where(t =>
                {
                    bool nonAbstractSerializable = !t.IsAbstract && (t.Attributes & TypeAttributes.Serializable) != 0;
                    ConstructorInfo ctorInfo = t.GetConstructor(BindingFlags.NonPublic |
                        BindingFlags.Instance, null, Type.EmptyTypes, null);

                    return (ctorInfo != null) ? nonAbstractSerializable && !ctorInfo.IsAssembly && !ctorInfo.IsFamily : nonAbstractSerializable;
                }); `